internal/locate/region_cache.go

```diff
@@ -1240,24 +1240,48 @@ func (c *RegionCache) BatchLocateKeyRanges(bo *retry.Backoffer, keyRanges []kv.K
 				keyRange.StartKey = lastRegion.EndKey()
 			}
 		}
+		// TODO: find all the cached regions in the range.
+		// now we only check if the region is cached from the lower bound, if there is a uncached hole in the middle,
+		// we will load the rest regions even they are cached.
+		r := c.tryFindRegionByKey(keyRange.StartKey, false)
```

Comment:
Can we remove this get single region?

Reply:
No. If we need to retrieve all regions by AscendGreaterOrEqual, since the Key of B-Tree is StartKey, we must retrieve the first region by SearchByKey.